src/main/java/apoc/convert/Convert.java

```diff
@@ -79,12 +88,91 @@ else if (list.getClass().isArray() && !list.getClass().getComponentType().isPrim
         }
         return null;
     }
-
+    
     @SuppressWarnings("unchecked")
+    private <T> List<T> convertToList(Object list, Class<T> type) {
+        List<Object> convertedList = convertToList(list);
+        if (convertedList == null) {
+        	return null;
+        }
+        Stream<T> stream = null;
+    	if (type.isAssignableFrom(Long.class)) {
+    		final Exact exact = new Exact();
+    		stream = (Stream<T>) convertedList.stream().map(e -> {
+    			if (e == null) {
+    				return null;
+    			}
+    			if (e instanceof Number) {
+    				return ((Number)e).longValue();
+    			}
+    			return exact.toInteger(toString(e), 0L, RoundingMode.HALF_UP.toString());
+    		});
+    	} else if (type.isAssignableFrom(Double.class)) {
+    		final Exact exact = new Exact();
+    		stream = (Stream<T>) convertedList.stream().map(e -> {
+				if (e == null) {
```

Comment:
In this case I'll use the [toDouble](https://github.com/neo4j-contrib/neo4j-apoc-procedures/blob/3.2/src/main/java/apoc/util/Util.java#L201) ok?